### PR TITLE
replaceWith

### PR DESCRIPTION
**Pull request by icanzilb**
as per https://github.com/RxSwiftCommunity/RxSwift-Ext/issues/8

pr includes docs, source, tests

@fpillet ?

### CODE REVIEW

File: Demo/RxSwiftExtDemo/RxSwiftExtPlayground.playground/Pages/replaceWith.xcplaygroundpage/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    
+    let numbers = Array<Int?>([1, 2, 3])
+    numbers.toObservable()
+        .replaceWith("candy")
+        .toArray()
+        .subscribeNext {result in
+            // look types on the right panel ===>
+            numbers
+            result
+            print("unwrap() transformed \(numbers) to \(result)")
```
Comment (fpillet) [inline Demo/RxSwiftExtDemo/RxSwiftExtPlayground.playground/Pages/replaceWith.xcplaygroundpage/Contents.swift]: Still mentions `unwrap`

File: Demo/RxSwiftExtDemo/RxSwiftExtPlayground.playground/Pages/replaceWith.xcplaygroundpage/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+/*:
+ ## replaceWith(AnyObject)
+ 
+ The `replaceWith` operator takes a sequence of elements and returns a sequence of the same constant provided as a parameter. In effect it ignores its input and replaces it with a constant
+ */
+example("replace any input with a value") {
+    
+    let numbers = Array<Int?>([1, 2, 3])
+    numbers.toObservable()
+        .replaceWith("candy")
```
Comment (fpillet) [inline Demo/RxSwiftExtDemo/RxSwiftExtPlayground.playground/Pages/replaceWith.xcplaygroundpage/Contents.swift]: Wot? replacing an `Int?` with a String ?

**Comment by icanzilb** [comment]
same same but different. @fpillet - the op is now `mapTo`

**Comment by fpillet** [comment]
👍

**Pull request merged**